### PR TITLE
[docker] Cleanup Docker Base-Deps

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?
This PR is currently a draft to reduce the number of times that `base-deps` is built. It doesn't make sense to constantly rebuild the image when it doesn't need to change all that often.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: ci/travis/build-docker-images.sh [hunk 1]
```diff
@@ -17,6 +17,24 @@ docker_push() {
...
 }
 
+build_or_pull_base_deps() {
+    docker pull rayproject/base-deps:latest
+    age=$(docker inspect -f '{{ .Created }}' rayproject/base-deps:latest)
+    tag=$(date +%F_%H)
+    # Build if older than 1 week or updated in this PR
+    if [[  $(date -d "-7 days" +%F) > $(date -d "$age" +%F) || "$RAY_CI_DOCKER_AFFECTED" == "1" ]]; then
+        docker image rm rayproject/base-deps:latest
+        docker build -t rayproject/base-deps docker/base-deps
```
Comment (simon-mo) [inline ci/travis/build-docker-images.sh:27]: `docker build --no-cache` so you don't need `docker image rm`

**Comment by ijrsvt** [comment]
@ijrsvt Make sure to add the docker hub repo and give the bot access.

File: docker/base-deps/Dockerfile [hunk 2]
```diff
@@ -31,9 +30,12 @@ RUN apt-get update -y && apt-get upgrade -y \
         flatbuffers \
         cython==0.29.0 \
         numpy==1.15.4 \
+        psutil \
+        blist \
```
Comment (simon-mo) [inline docker/base-deps/Dockerfile:34]: No longer required by serve. So it's safe to remove.

File: ci/travis/build-docker-images.sh [hunk 3]
```diff
@@ -16,6 +17,45 @@ docker_push() {
         echo "Skipping docker push because it's in PR environment."
     fi
 }
+build_and_push_tags() {
+    # $1 image-name (Dockerfile directory)
+    # $2 tag
+    for GPU in "" "-gpu" 
+    do 
+        BASE_IMAGE=$(if [ "$GPU" ]; then echo "nvidia/cuda:11.0-cudnn8-runtime-ubuntu18.04"; else echo "ubuntu:focal"; fi;)
+        SPECIFIC_NAME="rayproject/$1:$2$GPU"
```
Comment (simon-mo) [inline ci/travis/build-docker-images.sh:26]: It took me some time to grasp what it means. How about `FULL_NAME_WITH_TAG` and `LATEST_...`

File: ci/travis/build-docker-images.sh [hunk 4]
```diff
@@ -16,6 +17,45 @@ docker_push() {
         echo "Skipping docker push because it's in PR environment."
     fi
 }
+build_and_push_tags() {
+    # $1 image-name (Dockerfile directory)
```
Comment (simon-mo) [inline ci/travis/build-docker-images.sh:21]: Can we clearing this comment a little bit. I think you meant something like: image-name, we will also use this as the build directory for docker. So `$1=base-deps` means image will be called base-deps and we will build docker image in base-deps directory.

**Review by simon-mo — APPROVED** [review]
Mostly naming things. Great work!

File: ci/travis/build-docker-images.sh (hunk 3 above)
Comment (ijrsvt) [reply ci/travis/build-docker-images.sh:26]: Sounds good!

File: docker/base-deps/Dockerfile (hunk 2 above)
Comment (ijrsvt) [reply docker/base-deps/Dockerfile:34]: Fantastic!

**Pull request merged**